soc/nordic/nrf54l/Kconfig

```diff
@@ -27,6 +27,22 @@ config SOC_NRF54LX_SKIP_CLOCK_CONFIG
 	  With this option, the CPU clock frequency is not set during system initialization.
 	  The CPU runs with the default, hardware-selected frequency.
 
+config SOC_NRF54LX_DISABLE_FICR_TRIMCNF
+	bool "Disable trimming of the device"
+	default y if TRUSTED_EXECUTION_NONSECURE
+	help
+	  Disable trimming of the device. When the device is trimmed it
+	  will copy all the trimming values from FICR into the target
+	  addresses. When TF-M is enabled it will trim the device before
+	  booting this image.
+
+config SOC_NRF54LX_SKIP_GLITCHDETECTOR_DISABLE
+	bool "Skip disabling glitch detector"
+	default n
+	help
+	  With this option, the glitch detector is not disabled during system initialization.
+	  The CPU runs with the default state of glitch detector.
+
```

Comment:
The system initialization has this code:

NRF_GLITCHDET_S->bla bla;

_S is not available when TRUSTED_EXECUTION_NONSECURE.

We should prevent users from accidentally misconfiguring.

(TRUSTED_EXECUTION_NONSECURE && ! SOC_NRF54LX_SKIP_GLITCHDETECTOR_DISABLE) is invalid.

I'm not sure how best to express this in Kconfig.

One option would be to just change the default:

default y if TRUSTED_EXECUTION_NONSECURE

But that would only fix the default value.

Another option could be to remove the prompt:

bool
prompt "Skip disabling glitch detector" if ! TRUSTED_EXECUTION_NONSECURE
default y if TRUSTED_EXECUTION_NONSECURE

but that would cause issues for users that are explicitly configuring it with a Kconfig
fragment like:

CONFIG_bla_bla=y_or_n

Reply:
I suppose just changing the default value is good enough :)

Reply:
Changed